spec/rspec/rails/matchers/active_job_spec.rb

```diff
@@ -405,4 +405,261 @@ def self.name; "LoggingJob"; end
         .to have_been_enqueued.at(a_value_within(5.seconds).of(future))
     end
   end
+
+  describe "have_performed_job" do
+    before do
+      ActiveJob::Base.queue_adapter.perform_enqueued_jobs = true
+      ActiveJob::Base.queue_adapter.perform_enqueued_at_jobs = true
+
+      stub_const('HeavyLiftingJob', heavy_lifting_job)
+      stub_const('HelloJob', hello_job)
+      stub_const('LoggingJob', logging_job)
```

Comment:
Just curious what purpose this serves? We don't seem to use the constants so I'm curious...

Reply:
With `perform_enqueued_jobs` we're including ActiveJob's in the loop, that does its [`constantize` here](https://github.com/rails/rails/blob/af2129b/activejob/lib/active_job/core.rb#L51):
```ruby
job = job_data["job_class"].constantize.new
```
so we have to stub the constants.

Reply:
Copy, fancy adding a comment like this?
```
# stub_const is used so `job_data["job_class"].constantize` works
```

Reply:
Makes sense. Added.